habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_task.py

```diff
@@ -135,18 +128,68 @@ def reset(self, episode: Episode):
 
         # in the case of Stretch, force the agent to look down and retract arm with the gripper pointing downwards
         if isinstance(sim.robot, StretchRobot):
+            camera_rot = -1.7375 if self._pick_init else 0.0
             sim.robot.arm_motor_pos = np.array(
-                [0.0] * 4 + [0.775, 0.0, -1.57000005, 0.0, 0.0, -0.7125]
+                [0.0] * 4 + [0.775, 0.0, -1.57000005, 0.0, camera_rot, -0.7125]
             )
             sim.robot.arm_joint_pos = np.array(
-                [0.0] * 4 + [0.775, 0.0, -1.57000005, 0.0, 0.0, -0.7125]
+                [0.0] * 4 + [0.775, 0.0, -1.57000005, 0.0, camera_rot, -0.7125]
             )
 
+        start_hold_obj_idx: Optional[int] = None
+
+        # Only change the scene if this skill is not running as a sub-task
+        if (
+            self.force_obj_to_idx is None
+            and random.random() < self._config.object_in_hand_sample_prob
+            and self._goal_type != 'ovmm'
+        ):
+            start_hold_obj_idx = self._generate_snap_to_obj()
+
+        nav_to_pos = self._generate_nav_to_pos(
+            episode, start_hold_obj_idx=start_hold_obj_idx, force_idx=self.force_obj_to_idx
+        )
+
         self._nav_to_info = self._generate_nav_start_goal(
-            episode, force_idx=self.force_obj_to_idx
+            episode, nav_to_pos, start_hold_obj_idx=start_hold_obj_idx
         )
-        sim.robot.base_pos = self._nav_to_info.robot_start_pos
-        sim.robot.base_rot = self._nav_to_info.robot_start_angle
+        if self._pick_init:
```

Comment:
What does `_pick_init` signify?

Reply:
Added comments, testing a change where this is moved outside this file